fe1-web/model/network/method/message/Message.ts

```diff
@@ -130,14 +155,11 @@ export class Message {
     let privateKey = KeyPairStore.getPrivateKey();
     let signature: Signature;
 
-    // Get the properties of the type of message we want to sign
-    const messagesProperties = messagePropertiesMap.get(data.object)?.get(data.action);
-    if (messagesProperties === undefined) {
-      throw new Error(`Message signature for object ${data.object} and action ${data.action} is unsupported.`);
-    }
+    // Get the signature type of the type of message we want to sign
+    const signatureType = messageRegistry.getSignatureType(data);
 
     // If the message is signed with the pop token, get it from the store and sign the message
-    if (messagesProperties.isPopTokenSigned) {
+    if (signatureType === SignatureType.POP_TOKEN) {
```

Comment:
not for this PR, but I believe we could refactor the code from here to the end of the function. It's a bit repetitive and the `console.error` (line 168) doesn't abort/throw/return even though no meaningful work can be done after that error 😅

Reply:
That's true, I can do it.